src/taskgraph/docker.py

```diff
@@ -354,7 +355,47 @@ def _index(l: List, s: str) -> Optional[int]:
         pass
 
 
-def load_task(task_id: str, remove: bool = True, user: Optional[str] = None) -> int:
+def _resolve_image(image: Union[str, Dict[str, str]], graph_config: GraphConfig) -> str:
+    image_task_id = None
+
+    # Standard case, image comes from the task definition.
+    if isinstance(image, dict):
```

Comment:
Should we attempt to support the case where `payload.image` is just a string, like decision tasks?  It looks like currently that fails with `TypeError: string indices must be integers, not 'str'` because we assume it's a dict, and with this change it'd clash with the "refer to an in-tree image" case.  Maybe move that to `in-tree=foo` to leave the option open?

Reply:
That's https://github.com/taskcluster/taskgraph/pull/772/commits/39008139879e6a9fc298640702bb3e05be355cf4

But I'll just fold that into here as it's so trivial (just ended up separate due to how I happened to write my commits).

In this implementation the behaviour will be to first check for a local image, and then use a registry if there's no match, which I think I prefer to `in-tree=`, but don't feel super strongly about.